### PR TITLE
81 get single member page with all information user profile

### LINKED ISSUE #81: Get single member page with all information  (user profile)

### PR DESCRIPTION
**Pull request by mobatra**

### CODE REVIEW

**Review by Shatha-Amin — APPROVED** [review]

File: src/controllers/users/getSingleMemberController.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { CustomError, RequestWithDecoded, validateGetSingleMember } from '../../utils';
+
+const getSingleMemberController = async (
+  req: RequestWithDecoded,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { id } = req.params;
+    await validateGetSingleMember.validateAsync({ id });
```
Comment (Nour-Agha99) [inline src/controllers/users/getSingleMemberController.ts:12]: req.params always became string 
so make id number before to validate

File: src/controllers/users/getSingleMemberController.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    const { rows, rowCount } = await getSingleMemberQuery({ memberId: +id });
+
+    if (!rowCount) throw new CustomError('Member not found', 404);
+    res.status(200).json({
+      data: {
+        message: 'Member returned successfully',
```
Comment (Nour-Agha99) [inline src/controllers/users/getSingleMemberController.ts:19]: add error state in the response

File: src/controllers/users/getSingleMemberController.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const getSingleMemberController = async (
+  req: RequestWithDecoded,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { id } = req.params;
+    await validateGetSingleMember.validateAsync({ id });
+    const { rows, rowCount } = await getSingleMemberQuery({ memberId: +id });
```
Comment (Nour-Agha99) [inline src/controllers/users/getSingleMemberController.ts]: make label to rows to be more readable

**Review by Nour-Agha99 — CHANGES_REQUESTED** [review]

File: src/controllers/users/getSingleMemberController.ts (hunk 3 above)
Comment (mobatra) [reply src/controllers/users/getSingleMemberController.ts]: make it clear

File: src/database/query/users/getSingleMemberQuery.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    text: `
+      SELECT
+        users.full_name,
+        users.username,
+        users.email,
+        users.github,
+        users.linkedin,
+        users.avatar,
+        TO_CHAR(users.date_of_birth, 'YYYY-MM-DD') AS date_of_birth,
+        career_status.name AS career_status,
```
Comment (karamzomlot) [inline src/database/query/users/getSingleMemberQuery.ts:15]: No need to add `As career_status`. Just add it if you want to edit the column name when returning the data

**Review by KhaledAref13 — APPROVED** [review]

**Review by karamzomlot — CHANGES_REQUESTED** [review]

**Review by karamzomlot — APPROVED** [review]

**Review by Nour-Agha99 — APPROVED** [review]

**Pull request merged**